packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -636,4 +638,21 @@ export class ActionExecutionContext
 	async hackGetMediaObjectDuration(mediaId: string): Promise<number | undefined> {
 		return getMediaObjectDuration(this._context, mediaId, this.studioId)
 	}
+
+	async executeTSRAction(
+		deviceId: string,
+		actionId: string,
+		payload: Record<string, any>
+	): Promise<TSR.ActionExecutionResult> {
+		const device = await this._context.directCollections.PeripheralDevices.findOne({
+			_id: new RegExp(`${deviceId}$`),
+		})
```

Comment:
I don't think this is a good idea? If there is a device called `caspar01` and and another one called `backscreen_caspar01`, which one will be selected is undefined. I think this either needs to be an exact match (and the bluprints _just have to know_ what is the correct ID) or we need to have some sort of a middle step where the playoutGateway/mappings deviceId gets translated to a peripheralDeviceId.

Reply:
Maybe collaborate with @baltedewit on how to do the matching?

Reply:
I agree that we want this to be a strict match. Did you use a RegExp because it expects a Protected String otherwise? We have convenience methods for fixing that.

In addition I think we should add a timeout and perhaps instead of throwing an error we should return an execution result?